docs/admin/kubelet-tls-bootstrapping.md

```diff
@@ -0,0 +1,96 @@
+---
+assignees:
+- mikedanese
+
+---
+
+* TOC
+{:toc}
+
+## Summary
+
+This document describes setting up TLS client certificate bootstrapping for kubelets.
+Kubernetes 1.4 introduces an experimental API for requesting certificates from a cluster-level 
+Certificate Authority (CA). The first supported use of this API is the provisioning of TLS client 
+certificates for kubelets. The proposal can be found [here](https://github.com/kubernetes/kubernetes/pull/20439)
+and progress on the feature is being tracked as [feature #43](https://github.com/kubernetes/features/issues/43). 
+
+## apiserver configuration
+
+You must provide a token file which specifies at least one "bootstrap token" assigned to a kubelet boostrap-specific group.
+This group will later be used in the controller-manager configuration to scope approvals in the default approval
+controller. As this feature matures, you should ensure tokens are bound to an RBAC policy which limits requests
+using the bootstrap token to only be able to make requests related to certificate provisioning. When RBAC policy
+is in place, scoping the tokens to a group will allow great flexibility (e.g. you could disable a particular
+bootstrap group's access when you are done provisioning the nodes). 
+
+### Token auth file 
+Tokens are arbitrary but should represent at least 128 bits of entropy derived from a secure random number 
+generator (such as /dev/urandom on most modern systems). There are multiple ways you can generate a token. For example: 
+
+`head -c 16 /dev/urandom | od -An -t x | tr -d ' '` 
+
+will generate tokens that look like `02b50b05283e98dd0fd71db496ef01e8` 
+
+The token file will look like the following example, where the first three values can be anything and the quoted group 
+name should be as depicted: 
+
+```
+02b50b05283e98dd0fd71db496ef01e8,kubelet-bootstrap,10001,"system:kubelet-bootstrap"
```

Comment:
Do we have a set of authorization rules for this user group? I know that this requires the bootstrap token to only have access to specific sub resources (or it can auto approve its own request).